test/test_seqlock_writer_exclusion.c

```
@@ -0,0 +1,136 @@
+    printf("torn reads                : %" PRId64 "\n", torn);
+    printf("final seqlock             : %lu (%s)\n", final_seq,
+           (final_seq & 1) ? "ODD - a writer did not release" : "even");
+    printf("final total               : %lu (expected 0)\n", final_total);
+
+    if (torn != 0) {
+        fprintf(stderr, "FAIL: %" PRId64 " torn reads accepted under an even sequence\n",
+                torn);
+        return 1;
+    }
+    if (final_seq & 1) {
+        fprintf(stderr, "FAIL: sequence left odd\n");
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)